clients/venice-client/src/main/java/com/linkedin/venice/fastclient/ClientConfig.java

```diff
@@ -283,14 +296,20 @@ public ClusterStats getClusterStats() {
     private long routingUnavailableRequestCounterResetDelayMS = -1;
     private int routingPendingRequestCounterInstanceBlockThreshold = -1;
 
+    // TODO: 2 was set initially for singleGet based multiGet for a specific use case.
```

Comment:
This is a good question.
Today, the batch-get size enforcement is handled in Venice Router and it is configurable in store-level.
In Fast-Client, it is still an open question about how to setup the batch-get limit or whether we need any limit at all.
At the first step, It is better to keep the similar semantics, but enforced it in fast-client and later, we can adjust or remove this limit after a thorough validation.

Reply:
Also I don't think we should allow unlimited number of keys in a request. There should be a (configurable) upper limit. We aren't truly streaming at this point since we ask for all the keys in the request itself. Hence a limit will need to be applied or we will run into memory issues.

Reply:
Below items needs to be taken care in a new PR
1. the variable names for this limit is different for routers config and fast client. `maxKeyCountInMultiGetReq` vs `maxAllowedKeyCntInBatchGetReq`. Clean it up.
2. change it to behave like thinclient to start with: Set default to `500` as a global limit to start with and probably use per store limit as well.